rust/src/dataset.rs

```diff
@@ -87,12 +88,15 @@ async fn new_file_writer<'a>(
     object_store: &'a ObjectStore,
     data_file_path: &str,
     schema: &'a Schema,
+    max_group_size: usize,
```

Comment:
should we just always use the same variable name max_rows_per_group?
also it's weird that we intermix "group" and "batch"